### ISSUE TITLE
Undefined Symbol & 'GoogleUtilities not found' errors when building Xcode 12 project

### ISSUE THREAD

**Issue by ccamaisa** [firebase/firebase-ios-sdk#6284]
### [REQUIRED] Step 1: Describe your environment

  * Xcode version: Version 12.0 beta 3 (12A8169g)
  * Firebase SDK version: 6.30.0
  * Firebase Component: Firestore
  * Component version: 1.16.4
  * Installation method: CocoaPods

### [REQUIRED] Step 2: Describe the problem
I added Firebase to my project via cocoapods and built it, but I kept getting the error `Undefined symbol: _OBJC_CLASS_$_FIRApp`. 

After doing some searching, I found issue #6149. They seemed to be having the same issue. The workaround that was accepted was [this](https://github.com/firebase/firebase-ios-sdk/issues/6149#issuecomment-667201822), which I tried to no avail, even after OP's slight revision to the solution (see below)

Accepted workaround:
> Can you try adding -framework GoogleUtilities in your project's build settings under Other Linker Flags? This should attempt to force-include the framework and will give you a hopefully informative error if it's unable to.

OP response (worked for them):
> I added that flag (with a small change, I added "" around GoogleUtilities) and it compiles now

The error I encountered after that was `ld: framework not found GoogleUtilities`. 
I even tried adding it to the `podfile`, same error as well.

#### Steps to reproduce:

- Install Xcode beta
- Add Firebase/Firestore to project via cocoapods
- Add import/init code in AppDelegate
- compile 

#### Relevant Code:

AppDelegate:
```
import UIKit
import Firebase

@main
class AppDelegate: UIResponder, UIApplicationDelegate {



    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
        // Override point for customization after application launch.
        FirebaseApp.configure() // Configures Firebase
        return true
    }

    // MARK: UISceneSession Lifecycle

    func application(_ application: UIApplication, configurationForConnecting connectingSceneSession: UISceneSession, options: UIScene.ConnectionOptions) -> UISceneConfiguration {
        // Called when a new scene session is being created.
        // Use this method to select a configuration to create the new scene with.
        return UISceneConfiguration(name: "Default Configuration", sessionRole: connectingSceneSession.role)
    }

    func application(_ application: UIApplication, didDiscardSceneSessions sceneSessions: Set<UISceneSession>) {
        // Called when the user discards a scene session.
        // If any sessions were discarded while the application was not running, this will be called shortly after application:didFinishLaunchingWithOptions.
        // Use this method to release any resources that were specific to the discarded scenes, as they will not return.
    }


}
```

Podfile:
```
# Uncomment the next line to define a global platform for your project
# platform :ios, '9.0'

target 'ProjectName' do
  # Comment the next line if you don't want to use dynamic frameworks
  use_frameworks!

  # Pods for ProjectName
pod 'Firebase/Firestore'
pod 'FirebaseFirestoreSwift'
end
```

**Comment by morganchen12** [comment]
Normally this is something that CocoaPods will take care of for you. What version of CocoaPods are you using? Are you able to reproduce this build issue in the [Firestore quickstart](https://github.com/firebase/quickstart-ios/tree/master/firestore)?

**Comment by ccamaisa** [comment]
I'm currently using v.1.9.3. I'll try to reproduce with the that quickstart, will post result soon.

**Comment by ccamaisa** [comment]
@morganchen12 Just tried building the quickstart and it seems fine, also tried `pod deintegrate` and `pod clean` on my project yet it seems to still happen... the console even showed it downloading googleutilities 6.7.2

**Comment by ccamaisa** [comment]
@morganchen12 I did however create a new Xcode project on ver. 12 beta and added Firebase through the standard "Add Firebase to Your iOS App" process, still got the same error.

**Comment by morganchen12** [comment]
Can you share the new sample project you made? It's possible this is a CocoaPods bug and may be fixed in CocoaPods beta, but it'd be good to identify the root cause of the issue.

**Comment by paulb777** [comment]
Note that `_OBJC_CLASS_$_FIRApp` is defined in `FirebaseCore` not `GoogleUtilities`.

**Comment by ccamaisa** [comment]
My apologies, on that project I was getting the FirebaseApp.configure error() which was resolved (#6066). Implementing that fix worked.

Before, I added Firebase, where I got a plethora of errors and decided to go the cocoapods route. It looks like it messed with the project's internals and maybe wasn't removed properly. This is what I believe to be the case, but unfortunately I cannot share the original project for confidentiality reasons.

[CocoaPods pull #9905](https://github.com/CocoaPods/CocoaPods/pull/9905) shows that the issue I was experiencing was resolved in the latest beta of Cocoapods, which supports macOS 11. I have migrated my files to a new project and I haven't experienced any problems.
